### PR TITLE
[prometheus] Update to 2.43.0

### PR DESCRIPTION
**Pull request by nabokihms**
## Description
Bump Prometheus

## Why do we need it, and what problem does it solve?
Better performance, less bugs.

## What is the expected result?

Prometheus will be restarted.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: promtheus
type: fix
summary: Update Prometheus to 2.43.0 (bug and security fixes, performance improvements)
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Review by yalosev — APPROVED** [review]

File: modules/300-prometheus/images/prometheus/sample_limit_annotation.patch [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
+ 				podContainerNameLabel:  lv(c.Name),
+ 				podContainerIDLabel:    lv(cID),
+ 				podContainerImageLabel: lv(c.Image),
+@@ -313,6 +314,7 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
 
  			tg.Targets = append(tg.Targets, model.LabelSet{
  				model.AddressLabel:            lv(addr),
 +				"__sample_limit__":            lv(""),
  				podContainerNameLabel:         lv(c.Name),
- 				podContainerPortNumberLabel:   lv(ports),
```
Comment (yalosev) [inline modules/300-prometheus/images/prometheus/sample_limit_annotation.patch:11]: Why have we deleted these labels?

File: modules/300-prometheus/images/prometheus/sample_limit_annotation.patch [hunk 2]
```diff
@@ -59,7 +67,7 @@ index e017a4584..f3f7352ec 100644
  	"strings"
  	"sync"
  	"time"
-@@ -492,3 +493,15 @@ func TargetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig) ([]*Targe
+@@ -517,3 +518,15 @@ func TargetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig, noDefault
```
Comment (yalosev) [inline modules/300-prometheus/images/prometheus/sample_limit_annotation.patch:70]: seems weird, like function signature was cutted

Comment (yalosev) [reply modules/300-prometheus/images/prometheus/sample_limit_annotation.patch:70]: Ok. Build is successful, so nvm

File: modules/300-prometheus/images/prometheus/sample_limit_annotation.patch (hunk 1 above)
Comment (nabokihms) [reply modules/300-prometheus/images/prometheus/sample_limit_annotation.patch:11]: This is not deleted, the line is changed from 
```
podContainerPortNumberLabel: lv(ports),
```
to
```
podContainerIDLabel: lv(cID),
```
but not deleted.

See 
image

The + in the picture is connected to the patch, but the minus is connected to github diff.

**Pull request merged**